x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/documentation.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DocumentationProduct } from '@kbn/product-doc-common';
+import { FunctionVisibility } from '@kbn/observability-ai-assistant-plugin/common';
+import type { FunctionRegistrationParameters } from '.';
+
+export const RETRIEVE_DOCUMENTATION_NAME = 'retrieve_elastic_doc';
+
+export async function registerDocumentationFunction({
+  functions,
+  resources,
+  pluginsStart: { llmTasks },
+}: FunctionRegistrationParameters) {
+  const isProductDocAvailable = (await llmTasks.retrieveDocumentationAvailable()) ?? false;
```

Comment:
This is only called from `x-pack/plugins/observability_solution/observability_ai_assistant_app/server/plugin.ts`. When going from uninstalled => installed or vice versa, a server restart is required in order for the function to be included.

Reply:
Sorry not sure I get that? function registration in the o11y assistant framework is per-call, so this gets re-called / re-evaluated everytime the assistant has to answer.

Reply:
Got it, thanks